docs/java/java-testing.md

```diff
@@ -63,58 +134,68 @@ Sometimes you may want to customize the configuration to run your test cases. To
+- **sourcePaths**: Specify the extra source paths when debugging the tests
+- **vmArgs**: Specify the extra options and system properties for the JVM
+- **workingDirectory**: Specify the working directory when running the tests	
+
 More details can be found on the [vscode-java-test Wiki](https://github.com/Microsoft/vscode-java-test/wiki/Run-with-Configuration).
 
 ### View test results
-
 After running/debugging the test cases, the state of the related test items will be updated in both editor decorations and the Test Explorer.
 
 ![View test results](images/java-testing/test_report.png)
 
 You can trigger the command **Test: Peek Output** to peek the results view. You can select the links in the stack trace to navigate to the source location.
 
-### VS Code testing commands
+### Generate tests
+The extension provides features to help you scaffold test cases. You can find the entry in the editor context menu. Select `Source Action...` and then choose `Generate Tests...`.
 
-There are other testing commands (for example, **Run Tests in Current File**) that can be found by searching for 'Test:' in the Command Palette (`kb(workbench.action.showCommands)`).
+If you trigger this source action from your main source code (test subject), you will be asked the test class's fully qualified name and the methods you want to test, the extension will generate the test code for you:
 
-![Testing commands in the Command Palette](images/java-testing/command_palette.png)
-
-## Settings
+<video autoplay loop muted playsinline controls>
+  <source src="/docs/java/java-testing/generate-tests-from-main.mp4" type="video/mp4">
+</video>
 
-| Setting Name | Description | Default Value |
-|---|---|---|
-| `java.test.config` | Specify the configuration for the test cases to run with. [More details](https://aka.ms/java-test-config). | `{}` |
-| `java.test.defaultConfig` | Specify the name of the default test configuration. | `""` |
+If you trigger it from your test source code, you will be asked which kinds of test methods you want to add. Including the lifecycle methods and the test method:
 
-### VS Code testing settings
-
-There are VS Code settings specific to testing that can be found by searching for 'testing' in the Settings editor (`kb(workbench.action.openSettings)`).
+<video autoplay loop muted playsinline controls>
+  <source src="/docs/java/java-testing/generate-tests-from-test.mp4" type="video/mp4">
+</video>
 
-![Testing settings in the Settings editor](images/java-testing/settings.png)
+### Test navigation
+The extension provides features to help you navigate between your tests and test subjects. If your source code is contained in `src/main/java` or `src/test/java`, you can find the entry named `Go to Test` or `Go to Test Subject` in the editor context menu:
 
-## Project Setup
+![Go to tests](images/java-testing/goto-test.png)
 
-### JUnit 5
```

Comment:
should we also mention "java: go to test subject"command here?

Reply:
Maybe not, because the two command will both listed in the palette:

![image](https://user-images.githubusercontent.com/6193897/143815617-5557b2dc-b4a5-458e-8dac-a3a9530237d6.png)